Userland/Libraries/LibGfx/QOICommon.h

```diff
@@ -43,4 +45,35 @@ ALWAYS_INLINE static constexpr bool qoi_is_valid_run(u8 run)
     return AK::is_in_bounds(run, QOI_RUN_MIN, QOI_RUN_MAX);
 }
 
+class QOIState {
```

Comment:
We normally have each class in its own header.

Reply:
It was done to avoid the proliferation of dedicated headers. There are only a few shared component.